tools/serverpod_cli/lib/src/generator/dart/library_generators/model_library_generator.dart

```diff
@@ -49,6 +49,20 @@ class SerializableModelLibraryGenerator {
 
     return Library(
       (libraryBuilder) {
+        if (classDefinition.isSealedTopNode) {
+          for (var child in classDefinition.descendantClasses) {
+            libraryBuilder.directives
+                .add(Directive.part('${child.fileName}.dart'));
+          }
+        }
+
+        if (classDefinition.sealedTopNode != null &&
+            classDefinition.sealedTopNode != classDefinition) {
+          libraryBuilder.directives.add(
+            Directive.partOf('${classDefinition.sealedTopNode!.fileName}.dart'),
```

Comment:
Question: Why did you choose not to ask `!classDefinition.isSealedTopNode`?

Reply:
I did :) 

```dart
   if (!classDefinition.isSealedTopNode && sealedTopNode != null) {
          libraryBuilder.directives.add(
            Directive.partOf('${sealedTopNode.fileName}.dart'),
          );
        }
```

but we also have to check if `sealedTopNode != null` otherwise every class would have a part-of directive

Reply:
My bad, this was already fixed :)